src/engine.c

```diff
@@ -125,6 +125,66 @@ ENGINE_readFile(ENGINE* engine, const char* path, size_t* lengthPtr, char** reas
   return readEntireFile(pathBuf, lengthPtr, reason);
 }
 
+internal bool
+ENGINE_fileExists(ENGINE* engine, const char* path) {
+ char pathBuf[PATH_MAX];
+
+ if (strncmp(path, "./", 2) == 0) {
+   strcpy(pathBuf, path + 2);
+ } else {
+   strcpy(pathBuf, path);
+ }
+ 
+ if (engine->tar != NULL) {
+   bool tarCheck = fileExistsInTar(engine->tar, pathBuf);
+   if (tarCheck) {
+     return true;
+   }
+ }
+ 
+ if (path[0] != '/') {
+   strcpy(pathBuf, BASEPATH_get());
+   strcat(pathBuf, path);
+ }
+ return doesFileExist(pathBuf);
+}
+
+internal int
+ENGINE_directoryExists(ENGINE* engine, const char* path) {
+ char pathBuf[PATH_MAX];
+
+ if (strncmp(path, "./", 2) == 0) {
+   strcpy(pathBuf, path + 2);
+ } else {
+   strcpy(pathBuf, path);
+ }
+ 
+ if (engine->tar != NULL) {
```

Comment:
I changed my mind about how this should work.
Because the tar bundle is considered read-only, I don't think we need to check in the tar for existence before the disk. 'You' as the developer should know if a file or directory exists in there. If we want to check that in the future anyway (to handle modding or something) we can add that as an option later.

Reply:
Removed the tar bundle checks now directly checks the file system.